webpack/assets/javascripts/react_app/redux/API/APIMiddleware.js

```diff
@@ -1,11 +1,29 @@
+/* eslint-disable no-console */
 import { API_OPERATIONS, actionTypeGenerator } from './';
 import { get } from './APIRequest';
+import { startPolling } from './APIActions';
+import { selectPollingID } from './APISelectors';
+import { registeredPollingException } from './APIHelpers';
 
 export const APIMiddleware = store => next => action => {
-  const { type, key, payload = {}, url, actionTypes = {} } = action;
+  const { type, key, payload = {}, url, actionTypes = {}, polling } = action;
   if (type === API_OPERATIONS.GET) {
-    get(payload, url, store, actionTypeGenerator(key, actionTypes));
-  } else {
-    next(action);
+    const APIRequest = () =>
+      get(payload, url, store, actionTypeGenerator(key, actionTypes));
+    if (polling) {
+      const isAlreadyPolling = selectPollingID(store.getState(), key);
+      if (isAlreadyPolling) {
+        console.error(registeredPollingException(key));
+        return next(action);
+      }
+      /** 
+        force the APIRequest for the first time,
+        instead of waiting for the polling to start.
+      */
+      APIRequest();
+      return next(startPolling(key, APIRequest, polling));
```

Comment:
After some investigation, triggering `next(<another action>)` will not dispatch it from the beginning of the middlewares stack, so in this case, we need to use `dispatch`, which will run all the middlewares

Reply:
I couldn't find information about such an issue nor in the Redux middleware documentation,
could you explain more? from what I found returning `next(action)` worked as expected.

Reply:
after investigating more, and we talked about it offline, I removed the return and next
and instead use dispatch from the startPolling action. thanks @amirfefer

Reply:
`next(action)` will continue this action to the next middleware in the chain.
but prior middlewares won't be triggered, only `diaptch` will trigger all of the chain, so if it is a new action (like `startPolling`) and not the given action, some middlewares could be missed.